### ISSUE TITLE
DashboardLayout 100dvh

### ISSUE THREAD

**Issue by Zizico2** [mui/toolpad#4155]
### Search keywords

100dvh DashboardLayout

### Latest version

- [X] I have tested the latest version

### Summary

There's currently no way to have an element inside `DashboardLayout` fill the whole area, afaict. This is, personally, useful for a map. If `DashboardLayout`s height was `100vh`, this would be trivial. This could be set at the `DashboardLayout` level or at the `AppProvider` level, the latter would set it for all `DashboardLayout`s. If I attempt to set an element's height to `100vh`, it'll enable scroll, because it will add to the `AppBar`s height.

As some prior art, `Vuetify` sets `v-app`'s height to `100dvh` by default.

**Comment by Janpot** [comment]
I agree with this, it's a long standing issue I have with Our [`AppBar`](https://mui.com/material-ui/react-app-bar/) and [`Drawer`](https://mui.com/material-ui/react-drawer/) component. They display fixed and expect the body to be a scroll container.

I think we should instead make the `DashboardLayout` externally sizable (default 100vw,100vh?) and have a scroll container for the content only. This is essentially how we do layout in the Toolpad Studio editor and it feels a lot more natural.

**Comment by Zizico2** [comment]
Default `min-height/width` right? not `height/width`

**Comment by apedroferreira** [comment]
We can make the `DashboardLayout` height default to `100vh` and be overridable, I'm including the fix in https://github.com/mui/toolpad/pull/4119/commits/7998dce43b47d93d6b8e7e684b9a0fe0fdbd903e as it was necessary in that PR already

From what I tested it should be possible to include an element that fills the whole area after those changes!
